### PR TITLE
fix: feed `TransferProcess#properties` from the new field 

### LINKED ISSUE #3063: TransferProcess#properties wrongly populated
# Bug Report

## Describe the Bug
Currently `TransferProcess#properties` are populated by the` DataRequest#properties` but since the DataRequest it's stored it's a duplication of data. Originally `TransferProcess#properties` were introduced for storing the URL of dynamic edr receiver.  But those properties are transmitted over the dsp protocol. We should differentiate those two set of properties

## Possible Implementation
Take in input a new field in `TransferRequestDto` JsonObject that represent the "privateProperties" of a transfer process. Those properties will stay on the consumer side and they will not be transmitted to the counter party.

### PR DESCRIPTION
**Pull request by wolf4ood**
## What this PR changes/adds

Adds a new field `TransferProcessRequestDto#privateProperties` that will be stored in the `TransferProcess#properties` field

## Why it does that

Consistency

## Further notes

Also in this PR the `HttpDynamicEndpointDataReferenceReceiver` has been fixed

To make the change more consistent `TransferProcess#properties` has been renamed to `TransferProcess#privateProperties`

## Linked Issue(s)

Closes #3063 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md)._

### CODE REVIEW

File: core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public StatusResult<TransferProcess> initiateConsumerRequest(TransferRequest tra
                 .dataRequest(dataRequest)
                 .type(CONSUMER)
                 .clock(clock)
-                .properties(dataRequest.getProperties())
+                .properties(transferRequest.getPrivateProperties())
```
Comment (ndr-brt) [inline core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java:187]: why don't have a `privateProperties` field on `TransferProcess`? I'm thinking this in the process of getting rid of `DataRequest` (in fact `DataRequest` properties are `TransferProcess` properties)

Comment (ndr-brt) [reply core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java:187]: I think we wanted to get rid of `DataRequest` as a whole, because at the end it's just an unneeded type, so the `TransferProcess` `properties` are the `DataRequest` properties (as it is currently), 
while the private properties should be a completely new field.

TL;DR: `TransferProcess` should have a `properties` field and a `privateProperties` one.

maybe something for a subsequent PR, just wanted to check if we're aligned on this

Comment (wolf4ood) [reply core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java:187]: :+1:  in this case in this PR i would just rename `TransferProcess#properties` to `TransferProcess#privateProperties` then, so in subsequent PRs we can reintroduce `TransferProcess#properties` that will be the one from `DataRequest#properties`

Comment (ndr-brt) [reply core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java:187]: ok, fine for me

**Review by paullatzelsperger — APPROVED** [review]
LGTM, optionally we could rename the column in the Postgres schema. Pro: consistency, Contra: not strictly speaking needed, and it will cause migrations downstream.

**Review by ndr-brt — APPROVED** [review]
I'd rename the column as well, likely not a lot of people migrated to milestone-9 so they'll likely migrate to the new version directly, but this is already good for merge anyway
